hera_mc/mc_session.py

```diff
@@ -2588,9 +2589,17 @@ def correlator_control_command(self, command, starttime=None, duration=None,
                 # the correlator starttime can be different from the commanded
                 # time by as much as 134 ms
                 # the call to hera_corr_cm returns the actual start time (in unix format)
-                starttime_used_unix = \
-                    getattr(self.corr_obj, corr.command_dict[command])(starttime, duration, acclen, tag=tag)
-                starttime_used = Time(starttime_used_unix, format='unix')
+                # input time needs to be unix epoch in ms
+                starttime_ms = int(np.round(starttime.unix*1000))
+                starttime_used_unix_ms = \
+                    getattr(self.corr_obj, corr.command_dict[command])(starttime_ms, duration, acclen_spectra, tag=tag)
+                if starttime_used_unix_ms is hera_corr_cm.ERROR:
+                    warnings.warn('take_data returned ERROR on inputs: {t}, {d}, {acc},{tag}'.format(
```

Comment:
I think this should raise an error not a warning. If the command errors, it should not be recorded in the DB as a command that was successfully sent, which is what will happen with a warning. Also, line 2602 will error if the returned values is an error rather than a reasonable time.

Reply:
whats the kosher error type here?  RuntimeError?

Actually 2602 doesn't fail, it gives a time of 1-1-1970.

Reply:
Because the returned error value evaluates to False, which Time accepts for some reason.

Reply:
yes, I think RuntimeError is most appropriate.

Reply:
ok, added

Reply:
Wait, now you're raising a warning *and* an error? Why not just an error?

Reply:
Ok, now it just prints stuff in the error message.

(I originally left the warning because I thought warnings were piped somewhere, but on further investigation I don't think they are)